### ISSUE TITLE
Framework all: auto does not work if using <PascalCase /> component names

### ISSUE THREAD

**Issue by philfontaine** [quasarframework/quasar#5098]
**Describe the bug**
Framework all: auto does not work if using <PascalCase /> component names
e.g.
<QPage> ... </QPage> instead of <q-page> ... </q-page>

**Codepen/jsFiddle/Codesandbox (required)**
Fork a Codepen (https://codepen.quasar.dev) or a jsFiddle (https://jsfiddle.quasar.dev) or a Codesandbox (https://codesandbox.quasar.dev) and hit save then copy-paste link here.

https://codesandbox.io/s/codesandbox-app-deynv

**To Reproduce**
1. Rename a quasar component in PascalCase
This component is not registered, error in the browser
![image](https://user-images.githubusercontent.com/16233697/65167511-b905ae80-da10-11e9-9152-0bc6ae769fc1.png)

**Expected behavior**
The component should be registered correctly

**Screenshots**

**Platform (please complete the following information):**
OS:
Node:
NPM:
Yarn:
Browsers:
iOS:
Android:
Electron:

**Additional context**

**Comment by rstoenescu** [comment]
Hi,

This is by design, to keep the detection overhead as small as possible. I've got mixed feeling about adding pascal case detection too. I see the value in it, but worried about the perf implications.

**Comment by philfontaine** [comment]
I understand the decision, but I think it should be documented. 

On a side note, according to the [style guide](https://vuejs.org/v2/style-guide/#Component-name-casing-in-templates-strongly-recommended), PascalCase is the recommended casing in single-file components, although "using kebab-case everywhere is also acceptable".

Most people normally use one or the other, so the auto import could be something like `all: 'kebab'` or `all: 'pascal'`. This is just an idea to reflect on. I understand changing the API is not something you want to do often in a library.

Cheers

**Comment by bestog** [comment]
I have the same problem.
The idea from @philfontaine is good. :)

**Comment by rstoenescu** [comment]
Benchmark for capturing both camel-case and pascal-case:

<img width="144" alt="regex" src="https://user-images.githubusercontent.com/4665641/65247488-2260ec80-daf9-11e9-8291-68124e4f1a43.png">

27% slower. So will add a flag to pick what case to pick against.

**Comment by rstoenescu** [comment]
So, added support for another prop in quasar.conf.js > framework:

```js
all: 'auto',
autoImportComponentCase: 'kebab' // or 'pascal' or 'combined'
```

Kebab is default (you don't need to specify the new prop).
Pascal has equivalent perf with kebab.
Combined has the worse perf (~30-35% slower) than the other two, but supports both cases ('q-virtual-scroll' and 'QVirtualScroll').

Will be available in "@quasar/app" v1.1.2 --- and it will also require "quasar" v1.1.3.
